### PR TITLE
Implement operator= in ImageContainerByITKImage

### PR DESCRIPTION
**Pull request by phcerdan**
- [NA] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [NA] Doxygen documentation of the code completed (classes, methods, types, members...)
- [NA] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

**Comment by phcerdan** [comment]
Let me know if I can improve anything here. I think not having implemented it was just because nobody used it before. The base class defines the assignment operator, and all the other subclasses of ImageContainers have the operator implemented.

File: src/DGtal/images/ImageContainerByITKImage.ih [hunk 1]
```diff
@@ -152,6 +152,20 @@ namespace DGtal
...
+    template <typename TDomain, typename TValue>
+    inline
+    DGtal::ImageContainerByITKImage<TDomain,TValue>&
+    DGtal::ImageContainerByITKImage<TDomain,TValue>
+    ::operator=(const ImageContainerByITKImage& other)
+    {
+      if (this != &other)
+        {
+          myITKImagePointer = other.myITKImagePointer;
+          myDomain = other.myDomain;
```
Comment (rolanddenis) [inline src/DGtal/images/ImageContainerByITKImage.ih]: Should `updateDomain` be called here instead of copying domain, like in the copy constructor ?

**Comment by phcerdan** [comment]
Thanks @rolanddenis for the review, fixed, using `updateDomain()` as in the copy constructor.

**Comment by rolanddenis** [comment]
Seems good, merging 😉 Thanks !

**Pull request merged**